samza-core/src/main/java/org/apache/samza/table/remote/RemoteTable.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.table.remote;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.Validate;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.table.ReadableTable;
+import org.apache.samza.task.TaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Remote store backed readable table
+ *
+ * @param <K> the type of the key in this table
+ * @param <V> the type of the value in this table
+ */
+public class RemoteTable<K, V> implements ReadableTable<K, V> {
+  static final String READER_DATA = "io.reader";
+  static final String WRITER_DATA = "io.writer";
+
+  protected String tableId;
+  protected Logger logger;
+  protected RecordReader<K, V> recordReader;
+
+  public RemoteTable(String tableId, RecordReader recordReader) {
+    Validate.notBlank(tableId, "invalid table id");
+    Validate.notNull(recordReader, "null record reader");
+    this.tableId = tableId;
+    this.recordReader = recordReader;
+    this.logger = LoggerFactory.getLogger("RemoteTable-" + tableId);
+  }
+
+  @Override
+  public V get(K key) {
+    try {
+      return recordReader.get(key);
+    } catch (Exception e) {
+      logAndThrow(e, "Failed to get a record, key=%s", key);
+      return null;
+    }
+  }
+
+  @Override
+  public Map<K, V> getAll(List<K> keys) {
+    try {
+      return recordReader.getAll(keys);
+    } catch (Exception e) {
+      logAndThrow(e, "Failed to get records, keys=%s", keys);
+      return null;
+    }
+  }
+
+  @Override
+  public void init(Config config, TaskContext context) {
+    recordReader.init(config, context);
+  }
+
+  @Override
+  public void close() {
+    recordReader.close();
+  }
+
+  protected void logAndThrow(Exception cause, String format, Object ... args) {
```

Comment:
this is used by both RemoteTable and WritableRemoteTable. If we inline, we would have to write three lines instead of a single line (for methods without return value). 

Can you elaborate on the reasoning behind inlining?

Reply:
@pdu-mn1 : 
1. I am not entirely sure we are re-using much code by encapsulating `logAndThrow(..)` into its own 3 line function.
2. The combination of logAndThrow with a null return is confusing.
3. Personally, I'd prefer to get the entire view of what a function does when I read code as opposed to navigating another 3-line helper-method. Often, I'll have to ensure that the helper-function does not have un-intended side-effects, state mutations etc. Of course, there's not any established dogma here :-)